java/impl/aws/src/main/java/co/worklytics/psoxy/Handler.java

```diff
@@ -60,12 +89,67 @@ public APIGatewayV2HTTPResponse handleRequest(APIGatewayV2HTTPEvent httpEvent, C
                 .withStatusCode(response.getStatusCode())
                 .withHeaders(response.getHeaders())
                 .withBody(response.getBody())
-                .withIsBase64Encoded(false)
+                .withIsBase64Encoded(base64Encoded)
                 .build();
         } catch (Throwable e) {
             context.getLogger().log("Error writing response as Lambda return");
             throw new Error(e);
         }
     }
 
+    private boolean isCompressionRequested(HttpEventRequest request) {
+        return request.getHeader(HttpHeaders.ACCEPT_ENCODING).orElse("none").contains(ResponseCompressionHandler.GZIP);
+    }
+
+    static class ResponseCompressionHandler {
+
+        static final String GZIP = "gzip";
+        static final int DEFAULT_COMPRESSION_BUFFER_SIZE = 2048;
+        static final int MIN_BYTES_TO_COMPRESS = 2048;
+
+        @VisibleForTesting
+        boolean compressionOutweighOverhead(String body) {
+            return StringUtils.length(body) >= MIN_BYTES_TO_COMPRESS;
+        }
+
+        /**
+         * Compresses content as binary base64
+         * @param body
+         * @return optional with content if compression has been applied
+         */
+        Optional<String> compressBodyAndConvertToBase64(String body) {
```

Comment:
```suggestion
        Optional<String> compressBodyAndEncodeToBase64(String body) {
```

also, shouldn't this be `@VisibleForTesting`? I think intent is that the other method is the one called in practice from outside.